### ISSUE TITLE
zabbix-agent2 6.0 has broken dependencies, requires plugin repository

### ISSUE THREAD

**Issue by AnsgarKlein** [ansible-collections/community.zabbix#841]
##### SUMMARY

We have some problems with this role when installing zabbix-agent2 from repositories. It seems that zabbix-agent now requires the zabbix-plugins repository to be present, because newer versions of zabbix-agent depend on `zabbix-agent2-plugin-mongodb` and `zabbix-agent2-plugin-postgresql`.
Unfortunately this role only adds the "normal" Zabbix repository, not the "Zabbix-Plugin" Repository.

In my opinion this is more a general Zabbix Problem than a community.zabbix Ansible Problem but I reported [a Bug to Zabbix](https://support.zabbix.com/browse/ZBX-21940) and it seems to me like this is expected behaviour because the "proper" way to install zabbix-agent uses the zabbix-release package which _will_ add the Plugin Repository.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

/roles/zabbix_agent/tasks/Debian.yml: "Debian | Installing zabbix-agent"
/roles/zabbix_agent/tasks/RedHat.yml: "RedHat | Installing zabbix-agent"

##### ANSIBLE VERSION
```
ansible [core 2.13.6]
  config file = [...]
  configured module search path = ['~/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = ~[...]/venv/lib64/python3.8/site-packages/ansible
  ansible collection location = ~[...]/noventi.zabbix-agent/collections
  executable location = ~[...]/venv/bin/ansible
  python version = 3.8.12 (default, Sep 16 2021, 10:46:05) [GCC 8.5.0 20210514 (Red Hat 8.5.0-3)]
  jinja version = 3.1.2
  libyaml = True
```

##### CONFIGURATION

```
COLLECTIONS_PATHS(~[...]/ansible.cfg) = ['~[...]/collections']
DEFAULT_ROLES_PATH(~[...]/ansible.cfg) = ['~[...]/roles']
```

##### OS / ENVIRONMENT / Zabbix Version

I have discovered this problem on multiple operating systems (At least Ubuntu 20.04 and RHEL 8), but I have ran the tests to reproduce the issue in a clean Ubuntu 20.04 Vagrant box.
Ubuntu 20.04 and RHEL8 repositories currently contain version 6.0.10. I have systems running fine with zabbix-agent 6.0.7 without plugins packages installed.

I have tested this using community.zabbix 1.8.0 and main-branch.

##### STEPS TO REPRODUCE

Minimal test on Ubuntu 20.04 and RHEL8:

```yaml
- name: Install Zabbix Agent
  ansible.builtin.include_role:
    name: community.zabbix.zabbix_agent
  vars:
    zabbix_agent_become_on_localhost: false
    zabbix_repo: zabbix
    zabbix_agent_proxy: false
    zabbix_agent2: true
    zabbix_agent_version: '6.0'
    zabbix_agent_install_agent_only: true
    zabbix_http_proxy: '[...]'
    zabbix_https_proxy: '[...]'
```

##### EXPECTED RESULTS

zabbix-agent2 in version 6.0.* should be installed

##### ACTUAL RESULTS

Installing zabbix-agent2 fails because zabbix-agent2-plugin-postgresql and zabbix-agent2-plugin-mongodb are not available.
Those packages need to be installed from [Plugin-Repository](https://repo.zabbix.com/zabbix-agent2-plugins/) but this does not get installed by this Ansible role.

Ubuntu 20.04:

```
[...]

TASK [community.zabbix.zabbix_agent : Debian | Installing deb-src repository Ubuntu] ***
ok: [vagrant-test] => {"changed": false, "repo": "deb-src http://repo.zabbix.com/zabbix/6.0/ubuntu/ focal main", "state": "present"}

TASK [community.zabbix.zabbix_agent : Debian | Installing deb repository Ubuntu] ***
ok: [vagrant-test] => {"changed": false, "repo": "deb http://repo.zabbix.com/zabbix/6.0/ubuntu/ focal main", "state": "present"}

TASK [community.zabbix.zabbix_agent : Debian | Create /etc/apt/preferences.d/] ***
skipping: [vagrant-test] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [community.zabbix.zabbix_agent : Debian | Configuring the weight for APT] ***
skipping: [vagrant-test] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [community.zabbix.zabbix_agent : Debian | Installing zabbix-agent] ********
FAILED - RETRYING: [vagrant-test]: Debian | Installing zabbix-agent (3 retries left).
FAILED - RETRYING: [vagrant-test]: Debian | Installing zabbix-agent (2 retries left).
FAILED - RETRYING: [vagrant-test]: Debian | Installing zabbix-agent (1 retries left).
fatal: [vagrant-test]: FAILED! => {"attempts": 3, "cache_update_time": 1668617610, "cache_updated": true, "changed": false, "msg": "'/usr/bin/apt-get -y -o \"Dpkg::Options::=--force-confdef\" -o \"Dpkg::Options::=--force-confold\"       install 'zabbix-agent2=1:6.0.10-2+ubuntu20.04' -o APT::Install-Recommends=no' failed: E: Unable to correct problems, you have held broken packages.\n", "rc": 100, "stderr": "E: Unable to correct problems, you have held broken packages.\n", "stderr_lines": ["E: Unable to correct problems, you have held broken packages."], "stdout": "Reading package lists...\nBuilding dependency tree...\nReading state information...\nSome packages could not be installed. This may mean that you have\nrequested an impossible situation or if you are using the unstable\ndistribution that some required packages have not yet been created\nor been moved out of Incoming.\nThe following information may help to resolve the situation:\n\nThe following packages have unmet dependencies:\n zabbix-agent2 : Depends: zabbix-agent2-plugin-mongodb but it is not installable\n                 Depends: zabbix-agent2-plugin-postgresql but it is not installable\n", "stdout_lines": ["Reading package lists...", "Building dependency tree...", "Reading state information...", "Some packages could not be installed. This may mean that you have", "requested an impossible situation or if you are using the unstable", "distribution that some required packages have not yet been created", "or been moved out of Incoming.", "The following information may help to resolve the situation:", "", "The following packages have unmet dependencies:", " zabbix-agent2 : Depends: zabbix-agent2-plugin-mongodb but it is not installable", "                 Depends: zabbix-agent2-plugin-postgresql but it is not installable"]}

PLAY RECAP *********************************************************************
vagrant-test               : ok=16   changed=0    unreachable=0    failed=1    skipped=15   rescued=0    ignored=0
```

RHEL8:

```
[...]

TASK [community.zabbix.zabbix_agent : RedHat | Set zabbix_agent_distribution_major_version to 7 when Amazon 2] ***                                                                
skipping: [vagrant-test] => {"changed": false, "skip_reason": "Conditional result was False"}                                                                                     

TASK [community.zabbix.zabbix_agent : Fedora | Override zabbix_agent_distribution_major_version for Fedora <= 27] ***                                                             
skipping: [vagrant-test] => {"changed": false, "skip_reason": "Conditional result was False"}                                                                                     

TASK [community.zabbix.zabbix_agent : Fedora | Override zabbix_agent_distribution_major_version for Fedora >= 27] ***                                                             
skipping: [vagrant-test] => {"changed": false, "skip_reason": "Conditional result was False"}                                                                                     

TASK [community.zabbix.zabbix_agent : XCP-ng | Override zabbix_agent_distribution_major_version for XCP-ng] ***                                                                   
skipping: [vagrant-test] => {"changed": false, "skip_reason": "Conditional result was False"}                                                                                     

TASK [community.zabbix.zabbix_agent : RedHat | Install basic repo file] ********
changed: [vagrant-test] => (item={'name': 'zabbix', 'description': 'Zabbix Official Repository - $basearch', 'baseurl': 'https://repo.zabbix.com/zabbix/6.0/rhel/8/$basearch/', 'mode': '0644', 'gpgcheck': 0, 'gpgkey': 'file:///etc/pki/rpm-gpg/RPM-GPG-KEY-ZABBIX', 'state': 'present'}) => {"ansible_loop_var": "item", "changed": true, "item": {"baseurl": "https://repo.zabbix.com/zabbix/6.0/rhel/8/$basearch/", "description": "Zabbix Official Repository - $basearch", "gpgcheck": 0, "gpgkey": "file:///etc/pki/rpm-gpg/RPM-GPG-KEY-ZABBIX", "mode": "0644", "name": "zabbix", "state": "present"}, "repo": "zabbix", "state": "present"}                                                                                      
changed: [vagrant-test] => (item={'name': 'zabbix-non-supported', 'description': 'Zabbix Official Repository non-supported - $basearch', 'baseurl': 'https://repo.zabbix.com/non-supported/rhel/8/$basearch/', 'mode': '0644', 'gpgcheck': 0, 'gpgkey': 'file:///etc/pki/rpm-gpg/RPM-GPG-KEY-ZABBIX', 'state': 'present'}) => {"ansible_loop_var": "item", "changed": true, "item": {"baseurl": "https://repo.zabbix.com/non-supported/rhel/8/$basearch/", "description": "Zabbix Official Repository non-supported - $basearch", "gpgcheck": 0, "gpgkey": "file:///etc/pki/rpm-gpg/RPM-GPG-KEY-ZABBIX", "mode": "0644", "name": "zabbix-non-supported", "state": "present"}, "repo": "zabbix-non-supported", "state": "present"}          

TASK [community.zabbix.zabbix_agent : Do a yum clean] **************************
changed: [vagrant-test] => {"changed": true, "cmd": "yum clean all", "delta": "0:00:02.120961", "end": "2022-11-16 17:04:35.704725", "msg": "", "rc": 0, "start": "2022-11-16 17:04:33.583764", "stderr": "", "stderr_lines": [], "stdout": "Subscription Management Repositorys werden aktualisiert.\nLesen der Verbraucheridentität fehlgeschlagen\n\nDieses System ist nicht bei einem Berechtigungsserver registriert. Sie können sich mit dem Abo-Manager registrieren.\n\n0 Dateien entfernt", "stdout_lines": ["Subscription Management Repositorys werden aktualisiert.", "Lesen der Verbraucheridentität fehlgeschlagen", "", "Dieses System ist nicht bei einem Berechtigungsserver registriert. Sie können sich mit dem Abo-Manager registrieren.", "", "0 Dateien entfernt"]}

TASK [community.zabbix.zabbix_agent : RedHat | Installing zabbix-agent] ********
FAILED - RETRYING: [vagrant-test]: RedHat | Installing zabbix-agent (3 retries left).
FAILED - RETRYING: [vagrant-test]: RedHat | Installing zabbix-agent (2 retries left).
FAILED - RETRYING: [vagrant-test]: RedHat | Installing zabbix-agent (1 retries left).
fatal: [vagrant-test]: FAILED! => {"attempts": 3, "changed": false, "failures": [], "msg": "Depsolve Error occurred: \n Problem: cannot install the best candidate for the job\n  - nothing provides zabbix-agent2-plugin-mongodb needed by zabbix-agent2-6.0.10-release2.el8.x86_64\n  - nothing provides zabbix-agent2-plugin-postgresql needed by zabbix-agent2-6.0.10-release2.el8.x86_64", "rc": 1, "results": []}

RUNNING HANDLER [community.zabbix.zabbix_agent : clean repo files from proxy creds] ***

PLAY RECAP *********************************************************************
vagrant-test               : ok=11   changed=2    unreachable=0    failed=1    skipped=15   rescued=0    ignored=0 
```

**Comment by xbeastx** [comment]
+1 have same error:
```
Reading package lists...
Building dependency tree...
Reading state information...
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 zabbix-agent2 : Depends: zabbix-agent2-plugin-mongodb but it is not installable
                 Depends: zabbix-agent2-plugin-postgresql but it is not installable
                 ```

**Comment by xbeastx** [comment]
Temporary fix would be set `zabbix_apt_install_recommends: false`
But if you not using plugins of course...

**Comment by AnsgarKlein** [comment]
Thanks for the work around!
It seems this [will be fixed by Zabbix](https://support.zabbix.com/browse/ZBX-21855) for zabbix-agent 6.0

> I'm planning to add plugin packages to main 6.0 repository, to resolve the issue with LTS. (today)

**Comment by AnsgarKlein** [comment]
The dependencies have been fixed for 6.0 in the offizical repositories. This can be closed
